cli/resource.go

```diff
@@ -0,0 +1,298 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/odpf/salt/term" //nolint
+	entropyv1beta1 "go.buf.build/odpf/gwv/odpf/proton/odpf/entropy/v1beta1"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/odpf/salt/printer"
+	"github.com/spf13/cobra"
+)
+
+const outputJSON = "json"
+
+func cmdResource(ctx context.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "resource",
+		Aliases: []string{"resources"},
+		Short:   "Manage resources",
+		Annotations: map[string]string{
+			"group:core": "true",
+		},
+		Example: heredoc.Doc(`
+			$ entropy resource create
+			$ entropy resource list
+			$ entropy resource view
+			$ entropy resource delete
+			$ entropy resource edit
+		`),
+	}
+
+	cmd.AddCommand(
+		createResourceCmd(ctx),
+		listAllResourcesCommand(ctx),
+		viewResourceCommand(ctx),
+		editResourceCommand(ctx),
+		deleteResourceCommand(ctx),
+	)
+
+	return cmd
+}
+
+func createResourceCmd(ctx context.Context) *cobra.Command {
+	var filePath string
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "create a resource",
+		Example: heredoc.Doc(`
+			$ entropy resource create
+		`),
+		Annotations: map[string]string{
+			"action:core": "true",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+			cs := term.NewColorScheme()
+
+			var reqBody entropyv1beta1.CreateResourceRequest
+			if err := parseFile(filePath, &reqBody); err != nil {
+				return err
+			}
+			err := reqBody.ValidateAll()
+			if err != nil {
+				return err
+			}
+
+			client, cancel, err := createClient(cmd)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			res, err := client.CreateResource(ctx, &reqBody)
+			if err != nil {
+				return err
+			}
+			spinner.Stop()
+
+			fmt.Println("URN: \t", cs.Greenf(res.Resource.Urn))
```

Comment:
we should display the entire created resource here i think.

Reply:
`json.NewEncoder` and `yaml.NewEncoder` do not provide proper indentation. Any known packages which does the indenttaion for 'yaml' ? @spy16 @rohilsurana 

For, json we can still go with `json.MarshalIndent()` as discussed here as well https://github.com/odpf/entropy/pull/53#discussion_r884542550

Reply:
Sorry i don't understand. For JSON you can customise indentation using the encoder object methods.

YAML is indentation based format. So encoding involves indenting anyway right ?

Reply:
We should use `protojson` package instead of the standard `encoding/json` for [encoding protobuf messages](https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson#pkg-variables:~:text=This%20package%20produces%20a%20different%20output%20than%20the%20standard%20%22encoding/json%22%20package%2C%20which%20does%20not%20operate%20correctly%20on%20protocol%20buffer%20messages.).
Would suggest that we first convert protobuf to JSON and then from JSON to YAML.

We should use a similar flow when reading YAML from file.

Reply:
Thanks folks! Using the approach `first convert protobuf to JSON and then from JSON to YAML`